src/Kafka/KafkaInstances/components/InstancesTable.tsx

```diff
@@ -198,14 +204,20 @@ export const InstancesTable = <T extends KafkaInstance>({
               },
               {
                 title: t("table.actions.connection"),
-                onClick: () => onConnection(row),
+                ...(!changeConnectionEnabled
+                  ? {
+                      isDisabled: true,
+                    }
+                  : {
+                      onClick: () => onConnection(row),
+                    }),
```

Comment:
We have this kind of logic repeated a few times already in this component. It sounds like it's going to be a common pattern; maybe we should have a helper in the shared folder to generate these items.